frontend/catalyst/compiler.py

```diff
@@ -227,6 +227,19 @@ def get_output_filename(infile):
         return infile.replace(".buff.mlir", ".llvm.mlir")
 
 
+class JVPLoweringPass(PassPipeline):
+    """Pass pipeline to lower gradients."""
+
+    _executable = get_executable_path("quantum", "quantum-opt")
+    _default_flags = ["--lower-jvpvjp", "--debug-only=jvpvjp"]
+
+    @staticmethod
+    def get_output_filename(infile):
+        if not infile.endswith(".mlir"):
+            raise ValueError(f"Input file ({infile}) for quantum transforms is not an MLIR file")
+        return infile.replace(".mlir", ".novjp.mlir")
+
+
```

Comment:
Agreed, the pass can be part of the QuantumCompilation pass pipeline. I think it could even be included in the `--lower-gradients` pass by simply including the JVP patterns in that pass, unless there is a specific reason why one might want to lower `gradient.jvp` independently of `gradient.grad.

Reply:
I have joined the pass with the QuantumCompilationPass by adding an option. Let me run through one review cycle to understand is it fine to merge it further into the `--lower-gradients`

Reply:
One can always get the output of MLIR for debugging of individual passes (not pass pipelines) by using this flag `-mlir-print-ir-before-all`

Reply:
I suppose debugging is one reason, but the idea for the lower-gradients pass was to lower all gradients, whether it's jacobian, grad, jvp, or whatever else. Single op passes are not that typical I don't think.

Reply:
OK, now everything is merged into the `--lower-gradients` pass. While I was working on the code, I realized that the downside of this approach is that we no longer check the presence of `gradient.grad` operation during in the `JVPTest.mlir/VJPTest.mlir`, that might actually be helpful. So the ease of debug is definitely reduced now. Let me know if you think we must keep the `--lower-vjpvjp` as a separate pass.

Reply:
We solved this with pass options, you'll see in some tests we use `--lower-gradients=only=ps` for example. This allows you to select the individual patterns to apply (`fd`, `adj`, `ps`) for testing, while in general you can just run `--lower-gradients` and it applies all patterns. We should add `jvp` as an option to this list 🙂

Reply:
OK, done. It might be feasible to support comma-separated collections soon.